src/renderer/src/lib.rs

```diff
@@ -1,16 +1,204 @@
 #![crate_name = "amethyst_renderer"]
 #![crate_type = "lib"]
 #![doc(html_logo_url = "http://tinyurl.com/hgsb45k")]
+#![allow(dead_code)]
 
 //! High-level rendering engine with multiple backends.
 
+#[macro_use]
 extern crate gfx;
+#[macro_use]
+extern crate mopa;
 
-mod backend;
-mod frame;
-mod frontend;
-mod pipeline;
+extern crate glutin;
+extern crate cgmath;
 
-pub use self::frame::{Frame, FrameBuilder};
-pub use self::frontend::Renderer;
-pub use self::pipeline::{Pipeline, PipelineBuilder, Stage, Step};
+pub mod forward;
+mod gbuffer;
+mod wireframe;
+
+use std::any::TypeId;
+use std::collections::HashMap;
+
+use mopa::Any;
+
+pub use gbuffer::{GBuffer, Draw, BlitAmbiant, Lighting};
+
+pub type ColorFormat = gfx::format::Rgba8;
+pub type DepthFormat = gfx::format::DepthStencil;
+
+pub struct Renderer<R: gfx::Resources, C: gfx::CommandBuffer<R>> {
+    command_buffer: gfx::Encoder<R, C>,
+    methods: HashMap<(TypeId, TypeId), Box<Fn(&Box<Operation>, &Target, &Frame<R>, &mut gfx::Encoder<R, C>)>>
```

Comment:
uh oh, ok
this is apparently what forces you to do all these `R::* : Any` constraints when implementing `Method`